### PR TITLE
Update onnx models to have no weights

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Replace onnx models by the same models topology but having no weights. Add a function for loading such models and filling the weights by zero.

### Reason for changes

To reduce the repo size

### Related tickets

97804

### Tests

Tests were updated accordingly

### CODE REVIEW

**Comment by kshpv** [comment]
There is a problem with Mask RCNN model which has zero-detection with random weights.
It leads to an error during statistics collection.
Because of that this model can not be quantized

File: tests/onnx/quantization/helper.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    for tensor in onnx_graph.onnx_model.graph.initializer:
+        if uses_external_data(tensor):
+            np_dtype = TENSOR_TYPE_MAP[tensor.data_type].np_dtype
+            np_tensor = np.random.random(list(tensor.dims)).astype(np_dtype)
+            tensor_with_zeros = onnx.numpy_helper.from_array(np_tensor, name=tensor.name)
+            tensor.CopyFrom(tensor_with_zeros)
+    return model
+
+
+def save_model_without_tensors(model: onnx.ModelProto, model_filename: Path, model_dir: Path) -> None:
```
Comment (vshampor) [inline tests/onnx/quantization/helper.py:103]: Use https://docs.python.org/3/library/tempfile.html to work in a directory guaranteed to be temporary.

**Review by vshampor — CHANGES_REQUESTED** [review]
A great improvement! Still has a couple of things to be changed. Might want to add a README.md in `tests/onnx/data/models` describing that these modules are actually without weights and what to use in code to load and save such models.

File: tests/onnx/quantization/helper.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    :param model_path: Path to the onnx model to load.
+    :return: Onnx model with filled the all external tensors by random values.
+    """
+    model = onnx.load_model(model_path, load_external_data=False)
+    onnx_graph = ONNXGraph(model)
+    for tensor in onnx_graph.onnx_model.graph.initializer:
+        if uses_external_data(tensor):
+            np_dtype = TENSOR_TYPE_MAP[tensor.data_type].np_dtype
+            np_tensor = np.random.random(list(tensor.dims)).astype(np_dtype)
```
Comment (vshampor) [inline tests/onnx/quantization/helper.py:97]: Set seed before using random

File: tests/onnx/quantization/helper.py (hunk 1 above)
Comment (kshpv) [reply tests/onnx/quantization/helper.py]: Done

File: tests/onnx/quantization/helper.py (hunk 2 above)
Comment (kshpv) [reply tests/onnx/quantization/helper.py]: Replace to loading zeros weights, not random. It is better for degerming whether the model was loaded by provided method

File: tests/onnx/quantization/helper.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
+"""
```
[the remark this answers is not in the document]
Comment (kshpv) [reply tests/onnx/quantization/helper.py]: Done

File: tests/onnx/quantization/test_graphs.py [hunk 4]
```diff
@@ -13,15 +13,26 @@
...
 from tests.onnx.quantization.common import infer_model, min_max_quantize_model
 from tests.onnx.quantization.common import compare_nncf_graph
+from tests.onnx.quantization.common import mock_calculate_activation_quantizer_parameters
+from tests.onnx.quantization.common import mock_get_statistics
+from tests.onnx.quantization.common import mock_collect_statistics
 from tests.onnx.models import ALL_SYNTHETIC_MODELS
 from tests.onnx.models import MultiInputOutputModel
 
-
+@patch('nncf.quantization.algorithms.min_max.algorithm.calculate_activation_quantizer_parameters',
```
Comment (vshampor) [inline tests/onnx/quantization/test_graphs.py:28]: Use patching and mocking in the same manner as it is done in the majority of the rest of the tests. Search the code base for `mocker.patch` or `mocker.spy` for examples.

**Review by vshampor — CHANGES_REQUESTED** [review]

File: tests/onnx/quantization/test_graphs.py (hunk 4 above)
Comment (kshpv) [reply tests/onnx/quantization/test_graphs.py]: Done

**Review by KodiaqQ — APPROVED** [review]

**Review by l-bat — APPROVED** [review]

**Comment by ljaljushkin** [comment]
Will it actually reduce repository size? Shouldn't we carefully delete all these files from all previous commits to achieve that?
How did you measure it?

**Review by alexsu52 — APPROVED** [review]
LGTM

**Comment by kshpv** [comment]
@alexsu52 @ljaljushkin 
`nncf/tests/onnx/` size before: **1.4 GB** 
`nncf/tests/onnx/` size after: **7.3 MB**

**Comment by kshpv** [comment]
@alexsu52 can we merge this without @vshampor approve?

**Review by ljaljushkin — APPROVED** [review]

**Pull request merged**